ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/config/MergedApplicationConfig.kt

```diff
@@ -34,7 +34,8 @@ internal class MergedApplicationConfig(
         MergedApplicationConfig(first.config(path), second.config(path))
 
     override fun configList(path: String): List<ApplicationConfig> =
-        first.configList(path) + second.configList(path)
+        (if (firstKeys.contains(path)) first.configList(path) else emptyList()) +
```

Comment:
Could you please make it with a block body and extractions? so it can be easier to parse